compiler/rustc_lint/src/levels.rs

```diff
@@ -1028,6 +1030,7 @@ impl<'s, P: LintLevelsProvider> LintLevelsBuilder<'s, P> {
         struct_lint_level(self.sess, lint, level, src, span, msg, decorate)
     }
 
+    #[track_caller]
     pub fn emit_spanned_lint(
```

Comment:
Should this and similar have `#[rustc_lint_diagnostics]` also? it seems like there are a lot of lint-related diagnostic functions without `#[rustc_lint_diagnostics]`...

Reply:
It shouldn't matter, I don't even know if they trigger `-Ztreat-err-as-bug` which is presumably one of the main usages of tracking caller.

Reply:
edit: im dumb

`rustc_lint_diagnostics` shouldn't matter, ideally you'd add that to all the fns that we should deny outside of translatable diagnostics derives but that should be in a separate pr.